### PR TITLE
AdaptiveTimeSteppingEbos: move more code to compile unit

### PR DESCRIPTION
**Pull request by akva2**
One more micro.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — COMMENTED** [review]
Looks good for the most part, but you should `#include` the `ConvergenceReport.hpp` header wherever we use the actual `StepReport` type.

File: opm/simulators/flow/BlackoilModelEbos.hpp [hunk 1]
```diff
@@ -1061,13 +1061,6 @@ namespace Opm {
         const SimulatorReportSingle& failureReport() const
         { return failureReport_; }
 
-        struct StepReport
-        {
-            int report_step;
-            int current_step;
-            std::vector<ConvergenceReport> report;
-        };
```
[the remark this answers is not in the document]
Comment (akva2) [reply opm/simulators/flow/BlackoilModelEbos.hpp:1069]: fixed.

File: opm/simulators/flow/SimulatorFullyImplicitBlackoilEbos.hpp [hunk 2]
```diff
@@ -547,7 +547,7 @@ class SimulatorFullyImplicitBlackoilEbos
                      &this->convergenceOutputObject_.value());
     }
 
-    void writeConvergenceOutput(std::vector<typename Model::StepReport>&& reports)
+    void writeConvergenceOutput(std::vector<StepReport>&& reports)
```
[the remark this answers is not in the document]
Comment (akva2) [reply opm/simulators/flow/SimulatorFullyImplicitBlackoilEbos.hpp:551]: fixed.

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks for the updates.  I'll merge into master.

**Pull request merged**